lib/biran/config_defaults.rb

```diff
@@ -47,23 +47,35 @@ def config_dir
 
     def local_config_file
       ENV['BIRAN_LOCAL_CONFIG_FILE'] ||
-        File.join(app_shared_dir, configuration.config_dirname, configuration.local_config_filename)
+        File.join(app_shared_dir, configuration.config_dirname, local_config_filename)
+    end
+
+    def local_config_filename
+      ENV['BIRAN_LOCAL_CONFIG_FILENAME'] || app_config_defaults[:app][:local_config_filename] || configuration.local_config_filename
     end
 
     def vhost_public_dirname
       ENV['BIRAN_VHOST_PUBLIC_DIRNAME'] || app_config_defaults[:app][:vhost_public_dirname]
     end
 
     def db_config_override_file
-      File.join(app_shared_dir, configuration.config_dirname, configuration.db_config_filename)
+      File.join(app_shared_dir, configuration.config_dirname, db_config_filename)
+    end
+
+    def db_config_filename
+       app_config_defaults[:app][:db_config_filename] || configuration.db_config_filename
     end
 
     def secrets_file
-      File.join(configuration.base_path, configuration.config_dirname, configuration.secrets_filename)
+      File.join(configuration.base_path, configuration.config_dirname, secrets_filename)
+    end
+
+    def secrets_filename
+      app_config_defaults[:app][:secrets_filename] || configuration.secrets_filename
```

Comment:
I think that this can be optional I spend like two hours to figure out that on non rails projects you need set secrets

Reply:
That was a bug in biran that has been fixed but is not released yet. Has nothing to do with this as this IS optional. That bug was we expected the secrets hash to exist, so if it didn't, things would blow up. So now we set an empty hash on a few things so it doesn't blow up.